src/main/java/mate/academy/springboot/datajpa/service/CategoryServiceImp.java

```diff
@@ -0,0 +1,67 @@
+package mate.academy.springboot.datajpa.service;
+
+import javax.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.repository.CategoryRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImp extends BaseService<Category> implements CategoryService {
+
+    private final CategoryRepository repository;
+
+    @Override
+    @Transactional
+    public Category create(Category product) {
+        return repository.save(product);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Category getById(Long id) {
+        return repository.getById(id);
+    }
+
+    @Override
+    @Transactional
+    public Category update(Long id, Category source) {
+        Category target = repository.getById(id);
+        return update(source, target);
+    }
+
+    private Category update(Category source, Category target) {
+        if (target == null) {
+            return null;
+        }
+        if (source.getId() != null) {
+            target.setId(source.getId());
+        }
+        if (source.getDeleted() != null) {
+            target.setDeleted(source.getDeleted());
+        }
+        if (source.getName() != null) {
+            target.setName(source.getName());
```

Comment:
you don't need all these checks,  you don't even need an update method in this service, you can use create method, just rename it to save, you should pass already formed category